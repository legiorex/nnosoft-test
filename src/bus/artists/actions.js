// Types
import { types } from "./types";

// Instruments
export const artistsActions = {

    // Sync

    fillPosts: (artists) => {
        return {
            type:    types.FILL_POSTS,
            payload: artists,
        };
    },

    // Async

    fetchArtists: () => {
        return {
            type: types.FETCH_ARTISTS_ASYNC,
        };
    },
};
