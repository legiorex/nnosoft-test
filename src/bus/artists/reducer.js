// Core
import { fromJS, List } from 'immutable';

// Instruments
import { types } from "./types";

const initialState = [];

export const artistsReducer = (state = initialState, action) => {

    switch (action.type) {
        case types.FILL_ARTISTS:
            return action.payload;

        default:
            return state;
    }
};
