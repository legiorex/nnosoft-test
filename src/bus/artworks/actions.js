// Types
import { types } from "./types";

export const artworksActions = {

    // Sync

    fillArtworks: (artworks) => {
        return {
            type:    types.FILL_ARTWORKS,
            payload: artworks,
        };
    },

    // Async

    fetchArtworks: () => {
        return {
            type: types.FETCH_ARTWORKS_ASYNC,
        };
    },
};
