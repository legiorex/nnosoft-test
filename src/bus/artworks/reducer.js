// Components
import { types } from "./types";

const initialState = [];

export const artworksReducer = (state = initialState, action) => {

    switch (action.type) {
        case types.FILL_ARTWORKS:
            return action.payload;

        default:
            return state;
    }
};
