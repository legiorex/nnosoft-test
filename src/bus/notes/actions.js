// Types
import { types } from "./types";

export const notesActions = {

    // Sync

    addNote: (note) => {
        return {
            type:    types.ADD_NOTE,
            payload: note,
        };
    },

    writeNotesLocal: () => {
        return {
            type: types.WRITE_NOTES_LOCAL,
        };
    },
    sendNotes: (notes) => {
        return {
            type:    types.SEND_NOTES,
            payload: notes,
        };
    },

};
