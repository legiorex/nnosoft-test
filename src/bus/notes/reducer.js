// Components
import { types } from "./types";

const initialState = ["note 1", "note 2", "note 3", "note 4"];

export const notesReducer = (state = initialState, action) => {

    const stateUpdate = [];

    switch (action.type) {
        case types.ADD_NOTE:
            state.splice(Number(action.payload.indexNote) + 1, 0, action.payload.text);

            return stateUpdate.concat(state);

        case types.SEND_NOTES:
            return action.payload;
        default:
            return state;
    }
};
// action.payload.indexNote
// return [
//     ...state.slice(0, action.payload.indexNote + 1),
//     action.payload.text,
//     ...state.slice(action.payload.indexNote)
// ];
