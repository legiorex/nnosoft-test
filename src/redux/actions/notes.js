export const ADD_NOTE = "ADD_NOTE";

export function addNote (text, afterIndex) {
    return {
        type: ADD_NOTE,
        text,
        afterIndex,
    };
}
