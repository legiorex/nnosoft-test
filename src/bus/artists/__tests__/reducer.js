// Reducer
import { artistsReducer } from '../reducer';

// Actions
import { artistsActions } from '../actions';

describe('artists reducer:', () => {
    test('should return initial state by default', () => {
        expect(artistsReducer(void 0, {})).toMatchSnapshot();
    });

    test('should handle FILL_ARTISTS action', () => {
        expect(artistsReducer(void 0, artistsActions.fillArtists(__.artists))).toMatchSnapshot();
    });
});
