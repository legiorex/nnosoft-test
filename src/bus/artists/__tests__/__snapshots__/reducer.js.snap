// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`artists reducer: should handle FILL_ARTISTS action 1`] = `
Array [
  Object {
    "_id": "1",
    "firstName": "Annie",
    "lastName": "Walker",
    "photo": "https://www.url.com",
  },
]
`;

exports[`artists reducer: should return initial state by default 1`] = `Array []`;
